### ISSUE TITLE
Refined return types of getters breaks generated from method

### ISSUE THREAD

**Issue by marquiswang** [immutables/immutables#427]
If I create an interface

```
public interface Foo {
    Object foo();
}
```

and a Value class that implements it

```
@Value.Immutable
public interface Bar extends Foo {
    Double foo();
}
```

The generated code has a method:

```
    private void from(Object object) {
      long bits = 0;
      if (object instanceof Bar) {
        Bar instance = (Bar) object;
        if ((bits & 0x1L) == 0) {
          foo(instance.foo());
          bits |= 0x1L;
        }
      }
      if (object instanceof Foo) {
        Foo instance = (Foo) object;
        if ((bits & 0x1L) == 0) {
          foo(instance.foo());
          bits |= 0x1L;
        }
      }
    }
```

that doesn't compile because ```Foo#foo``` returns an Object and can't be set with ```Bar#foo```.

**Comment by elucash** [comment]
Please, check 2.2.12, we no longer copy such incompatible attributes. If all attributes from supertype are not compatible, we will not even generate `from` overload for it.

**Comment by marquiswang** [comment]
Yup, it is fixed in 2.2.12. Apologies and thanks!
